spark/src/main/scala/geotrellis/spark/summary/polygonal/RDDPolygonalSummary.scala

```diff
@@ -0,0 +1,149 @@
+        val uuid = UUID.randomUUID
+        val keys: Set[SpatialKey] = layout.mapTransform.keysForGeometry(geom)
+        keys.map(k => (k, Feature(geom, uuid)))
+      }
+
+    // Combine geometries and rasters by SpatialKey, then loop partitions
+    //  and perform the polygonal summary on each combination
+    val joinedRdd = keyedGeometryRdd.join(rasterRdd)
+    val featuresWithSummaries: RDD[(UUID, Feature[Geometry, R])] =
+      joinedRdd.map { data => {
+        val spatialKey = data._1
+        val feature: Feature[Geometry, UUID] = data._2._1
+        val tile: T = data._2._2
+        val extent: Extent = layout.mapTransform.keyToExtent(spatialKey)
+        val raster: Raster[T] = Raster(tile, extent)
+        val result: PolygonalSummaryResult[R] =
+          raster.polygonalSummary(feature.geom, visitor.getClass.newInstance, options)
+        result match {
+          case Summary(r: R) => (feature.data, Feature(feature.geom, r))
+          case NoIntersection => (feature.data, Feature(feature.geom, monoidR.empty))
+        }
+      }
+    }
+
+    // Geometries may have been split across SpatialKey leading to multiple UUID keys.
+    // Combine by UUID and then drop the UUID, which was added within this method
+    featuresWithSummaries
+      .reduceByKey {
+        case (a, b) =>
+          val summary = monoidR.combine(a.data, b.data)
+          Feature(a.geom, summary)
+      }
+      .map { _._2 }
+  }
+
+  trait RDDPolygonalSummaryMethods[R, T <: CellGrid[Int]]
+      extends MethodExtensions[RDD[(SpatialKey, T)] with Metadata[TileLayerMetadata[SpatialKey]]] {
+    def polygonalSummary[R](geometryRdd: RDD[Geometry],
+                            visitor: GridVisitor[Raster[T], R],
+                            options: Rasterizer.Options)(
+        implicit monoidR: Monoid[R], tagR: ClassTag[R]): RDD[Feature[Geometry, R]] = {
+      RDDPolygonalSummary(self, geometryRdd, visitor, options)
+    }
+
+    def polygonalSummary[R](geometry: Geometry,
+                            visitor: GridVisitor[Raster[T], R],
+                            options: Rasterizer.Options)(
+        implicit monoidR: Monoid[R], tagR: ClassTag[R], sc: SparkContext): RDD[Feature[Geometry, R]] = {
+      self.polygonalSummary(sc.parallelize(List(geometry)), visitor, options)
+    }
+
+    def polygonalSummary[R](geometry: Geometry,
+                            visitor: GridVisitor[Raster[T], R])(
+        implicit monoidR: Monoid[R], tagR: ClassTag[R], sc: SparkContext): RDD[Feature[Geometry, R]] = {
+      self.polygonalSummary(sc.parallelize(List(geometry)),
+                            visitor,
+                            PolygonalSummary.DefaultOptions)
+    }
+    def polygonalSummaryValue[R](geometry: Geometry,
+                                 visitor: GridVisitor[Raster[T], R],
+                                 options: Rasterizer.Options)(
+        implicit monoidR: Monoid[R], tagR: ClassTag[R], sc: SparkContext): R = {
```

Comment:
Hey, how come the result here is `R` and in raster case its `PolygonalSummaryResult[R]` wouldn't the same intersection concerns that required wrapping answer in `PolygonalSummaryResult` exist in this case as well?

Reply:
Going to point out that `RDD` case is not equivalent because `RDD` is a collection that can be empty, so in case of non-intersection an empty `RDD` is perfectly natural and maybe even expected.

Reply:
Resolving, this was addressed with some notes here in this comment: https://github.com/CloudNiner/geotrellis/pull/2#discussion_r296237778